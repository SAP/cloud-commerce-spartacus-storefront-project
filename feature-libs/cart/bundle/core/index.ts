export * from './connectors/index';
export * from './model/index';
export * from './bundle-core.module';
export * from './connectors/index';
export * from './facade/index';
export * from './store/actions/index';
export * from './store/selectors/index';
export * from './store/bundle-state';
export * from './services/index';
