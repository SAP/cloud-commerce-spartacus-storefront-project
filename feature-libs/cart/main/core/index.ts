export { CartCoreModule } from './cart-core.module';
export { CartPersistenceModule } from './cart-persistence.module';
export * from './connectors/index';
export * from './event/index';
export * from './facade/index';
export * from './services/index';
export * from './store/actions/index';
export * from './store/multi-cart-state';
export * from './store/selectors/index';
export * from './utils/utils';
