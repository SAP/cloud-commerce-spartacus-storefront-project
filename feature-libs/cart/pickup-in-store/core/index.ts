export * from './facades';
export * from './pickup-in-store-core.module';
