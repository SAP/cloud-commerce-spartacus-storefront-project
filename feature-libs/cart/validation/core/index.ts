export * from './cart-validation-core.module';
export * from './connectors/index';
export * from './facade/index';
export * from './model/index';
