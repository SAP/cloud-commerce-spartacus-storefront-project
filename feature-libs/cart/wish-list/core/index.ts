export * from './facade/index';
export * from './store/actions/index';
export * from './store/selectors/index';
export * from './store/wish-list-state';
export * from './utils/utils';
export * from './wish-list-core.module';
