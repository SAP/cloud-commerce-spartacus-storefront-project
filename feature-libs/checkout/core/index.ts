export * from './checkout-core.module';
