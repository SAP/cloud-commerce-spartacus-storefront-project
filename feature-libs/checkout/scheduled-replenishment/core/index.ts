export * from './checkout-scheduled-replenishment-core.module';
export * from './connectors/index';
export * from './facade/index';
