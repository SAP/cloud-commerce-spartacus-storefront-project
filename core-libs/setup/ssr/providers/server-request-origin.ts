import { inject } from '@angular/core';
import { SERVER_REQUEST_ORIGIN } from '@spartacus/core';
import { ServerOptions } from './model';

/**
 * Returns a factory function which resolves the server request origin.
 */
export function serverRequestOriginFactory(options?: ServerOptions): Function {
  return (): string => {
    const serverRequestOrigin = inject(SERVER_REQUEST_ORIGIN, {
      optional: true,
      skipSelf: true,
    });

    // SSR mode
    if (serverRequestOrigin) {
      return serverRequestOrigin;
    }

    // likely prerendering mode
    if (options?.serverRequestOrigin) {
      return options.serverRequestOrigin;
    }

    throw new Error(
      `The request origin is not set. Please specify it through environment variables when initiating the process.
    
    E.g.
    > SERVER_REQUEST_ORIGIN=https://my.domain.com yarn prerender
    > SERVER_REQUEST_ORIGIN=http://localhost:4200 yarn serve:ssr
    
    
    Alternatively, you can pass it as an argument to provideServer
    function, but beware it will be used for server-side rendering as well.
    
    E.g.
    @NgModule({
      // ...
      providers: [
        provideServer({
          serverRequestOrigin: 'https://my.domain.com',
        }),
      ],
    })
    export class AppServerModule {}`
    );
  };
}
