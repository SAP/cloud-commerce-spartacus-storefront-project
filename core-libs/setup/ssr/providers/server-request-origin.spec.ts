/*
 * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { TestBed } from '@angular/core/testing';
import { SERVER_REQUEST_ORIGIN } from '@spartacus/core';
import { reinitializeTestEnvironment } from '../../init-test-environment';
import { serverRequestOriginFactory } from './server-request-origin';

describe('serverRequestOriginFactory', () => {
  afterEach(() => {
    reinitializeTestEnvironment();
  });

  const mockOrigin = 'https://express.origin.com';

  describe('when SERVER_REQUEST_ORIGIN is provided in the platform injector', () => {
    beforeEach(() => {
      reinitializeTestEnvironment({
        platformProviders: [
          {
            provide: SERVER_REQUEST_ORIGIN,
            useValue: mockOrigin,
          },
        ],
      });
    });

    describe('and when options.serverRequestOrigin is NOT present', () => {
      beforeEach(() => {
        TestBed.configureTestingModule({
          providers: [
            {
              provide: SERVER_REQUEST_ORIGIN,
              useFactory: serverRequestOriginFactory({
                serverRequestOrigin: undefined,
              }),
            },
          ],
        });
      });

      it('should return SERVER_REQUEST_ORIGIN provided in the platform injector', () => {
        const result = TestBed.inject(SERVER_REQUEST_ORIGIN);
        expect(result).toEqual(mockOrigin);
      });
    });

    describe('and when options.serverRequestOrigin is present', () => {
      const configuredStaticOrigin = 'https://configured.origin.com';

      beforeEach(() => {
        TestBed.configureTestingModule({
          providers: [
            {
              provide: SERVER_REQUEST_ORIGIN,
              useFactory: serverRequestOriginFactory({
                serverRequestOrigin: configuredStaticOrigin,
              }),
            },
          ],
        });
      });

      it('should return options.serverRequestOrigin', () => {
        const result = TestBed.inject(SERVER_REQUEST_ORIGIN);
        expect(result).toEqual(configuredStaticOrigin);
      });
    });

    describe('when SERVER_REQUEST_ORIGIN is NOT present', () => {
      beforeEach(() => {
        reinitializeTestEnvironment({
          platformProviders: [
            // Note: no providers for SERVER_REQUEST_ORIGIN,
          ],
        });
      });

      describe('and when options.serverRequestOrigin is present', () => {
        const configuredStaticOrigin = 'https://configured.origin.com';

        beforeEach(() => {
          TestBed.configureTestingModule({
            providers: [
              {
                provide: SERVER_REQUEST_ORIGIN,
                useFactory: serverRequestOriginFactory({
                  serverRequestOrigin: configuredStaticOrigin,
                }),
              },
            ],
          });
        });

        it('should return options.serverRequestOrigin', () => {
          const result = TestBed.inject(SERVER_REQUEST_ORIGIN);
          expect(result).toEqual(configuredStaticOrigin);
        });
      });

      describe('and when options.serverRequestOrigin is NOT present', () => {
        beforeEach(() => {
          TestBed.configureTestingModule({
            providers: [
              {
                provide: SERVER_REQUEST_ORIGIN,
                useFactory: serverRequestOriginFactory({
                  serverRequestOrigin: undefined,
                }),
              },
            ],
          });
        });

        it('should throw an error', () => {
          expect(() =>
            TestBed.inject(SERVER_REQUEST_ORIGIN)
          ).toThrowErrorMatchingSnapshot();
        });
      });
    });
  });
});
