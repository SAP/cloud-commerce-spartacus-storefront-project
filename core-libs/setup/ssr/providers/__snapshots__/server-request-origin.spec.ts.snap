// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serverRequestOriginFactory when SERVER_REQUEST_ORIGIN is provided in the platform injector when SERVER_REQUEST_ORIGIN is NOT present and when options.serverRequestOrigin is NOT present should throw an error 1`] = `
"The SERVER_REQUEST_ORIGIN is not provided. 
    If you are using the default environment variable, please specify it when initiating the process.
    
    E.g.
    > SERVER_REQUEST_ORIGIN=https://my.domain.com yarn prerender
    > SERVER_REQUEST_ORIGIN=http://localhost:4200 yarn serve:ssr
    
    
    Alternatively, you can pass it as an argument to 'provideServer()'
    function, but beware it will be used for server-side rendering as well.
    
    E.g.
    @NgModule({
      // ...
      providers: [
        provideServer({
          serverRequestOrigin: 'https://my.domain.com',
        }),
      ],
    })
    export class AppServerModule {}"
`;
